src/coreclr/tools/Common/TypeSystem/IL/UnsafeAccessors.cs

```diff
@@ -563,6 +820,15 @@ private static MethodIL GenerateAccessorSpecificFailure(ref GenerationContext co
                 codeStream.EmitLdc((int)ExceptionStringID.InvalidProgramDefault);
                 thrower = typeSysContext.GetHelperEntryPoint("ThrowHelpers", "ThrowInvalidProgramException");
             }
+            else if (result is SetTargetResult.NotSupported)
+            {
+                thrower = typeSysContext.GetHelperEntryPoint("ThrowHelpers", "ThrowNotSupportedException");
+            }
+            else if (isSignatureLoad)
```

Comment:
Can we encode `MissingType` as a `SetTargetResult`? The extra `bool` here seems very odd.